sentry/sentry/nat_option.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2020-2022 The Silkworm Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "nat_option.hpp"
+#include <boost/algorithm/string/predicate.hpp>
+
+namespace silkworm::sentry {
+
+bool lexical_cast(const std::string& input, NatOption& value) {
+    if (input == "none") {
+        value = {};
+        return true;
+    }
+    if (boost::algorithm::istarts_with(input, "extip:")) {
+        std::string ip_str = input.substr(6);
+        boost::system::error_code err;
+        auto ip = boost::asio::ip::address::from_string(ip_str, err);
```

Comment:
Why copy allocate a new string instead of consuming current one ?

```suggestion
        input.erase(0,6);
        boost::system::error_code err;
        auto ip = boost::asio::ip::address::from_string(input, err);
```
This obviously implies removing `const` from input.

Also I believe this
`auto ip = boost::asio::ip::address::from_string(ip_str, err);`
is meant to fail if input is `xxx.xxx.xxx.xxx:yyyyy` where y are placeholders for port number

Reply:
I got it fixed in a different way, because such parameter modification would be an unexpected side effect from the caller's point of view.
I'd rather use something like a string_view here, but this boost API can't deal with it.

As for the port, yes, it does fail on such an input as expected. The option only accepts the IP, no port is allowed.